### ISSUE TITLE
CotEditor crash when opening preference pane to 'appearance' with specific font settings

### ISSUE THREAD

**Issue by spencerwooo** [coteditor/CotEditor#1647]
### Description

After upgrading CotEditor to the latest version (4.8.4 (651)) the editor consistently crashes when opening preferences and navigating to the 'appearance' panel (this panel shown in the screenshot).

<img src="https://github.com/coteditor/CotEditor/assets/32114380/7c040449-ab4c-41a0-a159-87ec9075e241" alt="screenshot" width="512" >

### To Reproduce

_I do not know if this issue can be reproduced._ I will document how I fixed this on my side.

Before upgrading and experiencing the crash, my CotEditor have the following settings:

```console
$ defaults read com.coteditor.CotEditor fontName
JetBrainsMonoNF-Regular

$ defaults read com.coteditor.CotEditor fontSize
14
```

After upgrading CotEditor and the crashed happened, I tried to perform a fresh install by removing all CotEditor related files:

- `/Users/spencerwoo/Library/Containers/com.coteditor.CotEditor`
- `/var/db/receipts/com.coteditor.CotEditor.bom`

It did not work and CotEditor still crashes.

I then removed all CotEditor `defaults` by:

```console
$ defaults delete-all com.coteditor.CotEditor
```

which seems to have solved this crash.

### Expected behavior

CotEditor not crashing when opening preferences - appearance.

### CotEditor version

4.8.4 (651)

### macOS version

14.5 (23F79)

### Additional context

I have now set the fonts of CotEditor to the system preinstalled 'Menlo' in fear of the crash happening again 😢 

Additionally, the `defaults` of CotEditor at its current form is:

```console
$ defaults read com.coteditor.CotEditor
{
    NSFontPanelAttributes = "1, 0";
    "NSSplitView Subview Frames WindowContentSplitView" =     (
        "0.000000, 0.000000, 814.000000, 767.000000, NO, NO",
        "814.000000, 0.000000, 220.000000, 767.000000, YES, NO"
    );
    "NSToolbar Configuration Document" =     {
        "TB Display Mode" = 2;
        "TB Icon Size Mode" = 1;
        "TB Is Shown" = 1;
        "TB Size Mode" = 1;
    };
    "NSWindow Frame Document" = "1266 304 814 767 0 0 2560 1415 ";
    "NSWindow Frame NSFontPanel" = "2045 159 445 239 0 0 2560 1415 ";
    defaultTheme = Anura;
    font = {length = 500, bytes = 0x62706c69 73743030 d4010203 04050607 ... 00000000 0000017e };
    lastPreferencesPaneIdentifier = appearance;
    lineHeight = "1.3";
    monospacedFont = {length = 500, bytes = 0x62706c69 73743030 d4010203 04050607 ... 00000000 0000017e };
    pinsThemeAppearance = 0;
    recentStyleNames =     (
        "Plain Text"
    );
    snippets =     (
    );
}
```

Unfortunately I have no record of `defaults read com.coteditor.CotEditor` when the crash is still reproducible.

**Comment by 1024jp** [comment]
This issue was already reported at #1644 and will be fixed on CotEditor 4.8.5, which has already been submitted to the App Store and will be released within a day.

**Comment by spencerwooo** [comment]
Sorry I was not aware of that issue when searching for similar ones. Glad that its fixed :)
